src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -961,10 +956,11 @@ private DateTime AddUnits(double value, long maxUnitCount, long ticksPerUnit)
         // or equal to d that denotes a valid day in month m1 of year
         // y1.
         //
-        public DateTime AddMonths(int months)
+        public DateTime AddMonths(int months) => AddMonths(this, months);
+        private static DateTime AddMonths(DateTime date, int months)
```

Comment:
This is not a good practice. The static method doesn't save anything.

Reply:
The calling convention for struct instance methods always forces the struct on stack, unless the method is inlined.

Reply:
I see the problem in calling convention. Is there any measurement about the performance impact?